### PR TITLE
Nostr processing

### PR DESCRIPTION
**Pull request by tompro**
- Processing Nostr messages in a separate thread 
- Added Nostr relay mock for unit testing (this is flaky if we want to add more transport test see comment)
- E2E test with send -> serialize -> receive -> unserialize -> handle

Looking for ways to test Nostr private messages I found out that yes we have the latest cool shit (NIP17 private messaging) but most prominent relays out in the wild do not support this yet (there are a few paid ones that already do). It could be that for a while we have to run our own to make E2E manual testing easier in BE and FE.

### CODE REVIEW

File: src/service/notification_service/nostr.rs [hunk 1]
```diff
@@ -60,12 +97,220 @@ impl NotificationJsonTransportApi for NostrClient {
...
+                            if contact_service.is_known_npub(sender.as_str()).await? {
+                                trace!("Received event: {envelope:?} from {sender:?}");
+                                handle_event(envelope, &event_handlers).await?;
+                            }
+                        }
+                    };
+                    Ok(false)
+                })
+                .await
+                .expect("Nostr notification handler failed");
```
Comment (zupzup) [inline src/service/notification_service/nostr.rs:173]: Should this restart, or something along those lines if it fails, or is it fine, if the app crashes in this case?
We should probably think about that for the dht part as well :thinking:

**Review by zupzup — APPROVED** [review]
LGTM :+1: 

Nostr integration coming along nicely! :muscle:

File: src/bill/identity.rs [hunk 2]
```diff
@@ -47,6 +47,7 @@ pub struct Identity {
     pub bitcoin_public_key: String,
     pub bitcoin_private_key: String,
     pub nostr_npub: Option<String>,
+    pub nostr_relay: Option<String>,
```
[the remark this answers is not in the document]
Comment (tompro) [reply src/bill/identity.rs:50]: That is the whole thing about Nostr relays, everybody can use the one they prefer. That in turn means clients have to communicate one or a bunch of relays where they expect notes to be sent to (inbox) as we as all incoming notes to have a relay where clients should send a reply to.

[the remark this answers is not in the document]
Comment (mtbitcr) [reply src/bill/identity.rs:50]: In case I use multiple relays, how will it work? Because I see that we don't have a vector of strings, but only one. Or?

**Review by mtbitcr — APPROVED** [review]

**Comment by tompro** [comment]
@mtbitcr yes I was thinking about a relay list as well. For DMs in Nostr you should always send to the Inbox relay where you usually have a single one configured (but could be multiple). I chose to use a single one just bcs it is way simpler to keep in sync in the contacts when we pull them P2P. We can still switch it to a relay list later if we discover it is necessary :thinking: .

**Pull request merged**